crates/wasi/tests/all/p3/mod.rs

```diff
@@ -0,0 +1,99 @@
+#![cfg(feature = "p3")]
+
+use std::path::Path;
+
+use anyhow::{Context as _, anyhow};
+use wasmtime::Store;
+use wasmtime::component::{Component, Linker, ResourceTable};
+use wasmtime_wasi::p3::bindings::Command;
+use wasmtime_wasi::p3::{WasiCtx, WasiCtxBuilder, WasiView};
+use wasmtime_wasi::{DirPerms, FilePerms};
+
+use test_programs_artifacts::*;
+
+macro_rules! assert_test_exists {
+    ($name:ident) => {
+        #[expect(unused_imports, reason = "just here to assert it exists")]
+        use self::$name as _;
+    };
+}
+
+struct Ctx {
+    ctx: WasiCtx,
+}
+
+impl WasiView for Ctx {
+    fn ctx(&mut self) -> &mut WasiCtx {
+        &mut self.ctx
+    }
+}
+
+// TODO: Remove once test components are not built for `wasm32-wasip1`
+impl wasmtime_wasi::p2::WasiView for Ctx {
+    fn ctx(&mut self) -> &mut wasmtime_wasi::p2::WasiCtx {
+        panic!("should not use wasip2")
+    }
+}
+
+// TODO: Remove once test components are not built for `wasm32-wasip1`
+impl wasmtime_wasi::p2::IoView for Ctx {
+    fn table(&mut self) -> &mut ResourceTable {
+        panic!("should not use wasip2")
+    }
+}
```

Comment:
Could these get filled out to assist with debugging? That way `println!` in tests should work and such I believe. It's fine to leave the `TODO` comments though because it's true that yes, ideally, the components don't use WASIp2 at all for tests here (but there's a long way to get there)